### PR TITLE
[FEATURE] Cloud support FDS deletes

### PR DESCRIPTION
**Pull request by Kilo59**
## Changes proposed in this pull request:
- Call `self._datasource_store.delete()` when `_delete_fluent_datasource` is called.
- Update `get_datasource` to work with FluentDatasources when pulling from cloud
- Update `DatasourceStore.ge_cloud_response_json_to_object_dict()` to handle an array of datasource objects

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -756,6 +756,9 @@ def _update_fluent_datasource(
         else:
             datasource_name = kwargs.get("name", "")
 
+    def _update_fluent_datasource(self, datasource: FluentDatasource) -> None:
+        # NOTE: this is not called
+        self.datasources[datasource.name] = datasource
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py:761]:
```suggestion
```

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 2]
```diff
@@ -776,8 +776,14 @@ def _update_fluent_datasource(
 
         self.datasources[datasource_name] = update_datasource
 
-    def _delete_fluent_datasource(self, datasource_name: str) -> None:
-        self.datasources.pop(datasource_name, None)
+    def _delete_fluent_datasource(
+        self, datasource_name: str, _call_store: bool = True
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py:780]: Alternate param name `local_only`, `skip_store`, `persist_to_store` etc.

Comment (Kilo59) [reply great_expectations/data_context/data_context/abstract_data_context.py:780]: Explain this

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 3]
```diff
@@ -776,7 +776,20 @@ def _update_fluent_datasource(
...
+    ) -> None:
+        """
+        _call_store = False allows for local deletes without deleting the persisted storage datasource.
+        This should generally be avoided.
+        """
+        datasource = self.datasources.get(datasource_name)
+        if datasource:
+            if self._datasource_store.cloud_mode and _call_store:
+                self._datasource_store.delete(datasource)  # type: ignore[arg-type] # Could be a LegacyDatasource
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py:789]: We have work scoped to use a new `FluentDatasourceStore` for all FDS persistence logic and avoid this awful branching.

File: tests/datasource/fluent/test_contexts.py [hunk 4]
```diff
@@ -26,6 +26,13 @@
 yaml = YAMLHandler()
 
 
+@pytest.fixture
+def taxi_data_samples_dir() -> pathlib.Path:
+    return pathlib.Path(
+        __file__, "..", "..", "..", "test_sets", "taxi_yellow_tripdata_samples"
+    ).resolve(strict=True)
```
Comment (cdkini) [inline tests/datasource/fluent/test_contexts.py:33]: Does `resolve` confirm existence? If not, could we do an `exists()` check?

File: tests/datasource/fluent/test_contexts.py [hunk 5]
```diff
@@ -100,35 +109,81 @@ def test_assets_are_persisted_on_creation_and_removed_on_deletion(
...
+        response.raise_for_status()
+        print(pf(response.json(), depth=4))
+        pytest.xfail(  # TODO: remove this
+            "Immediate persistance for cloud not implemented"
+        )
+        assert response.json()["data"]["attributes"]["datasource_config"].get("assets")
+
+
+@pytest.mark.cloud
+def test_cloud_context_delete_datasource(
```
Comment (cdkini) [inline tests/datasource/fluent/test_contexts.py:151]: `@pytest.mark.integration`?

**Review by cdkini — APPROVED** [review]

File: tests/datasource/fluent/test_contexts.py (hunk 4 above)
Comment (Kilo59) [reply tests/datasource/fluent/test_contexts.py:33]: With strict=True it does

File: tests/datasource/fluent/test_contexts.py (hunk 5 above)
Comment (Kilo59) [reply tests/datasource/fluent/test_contexts.py:151]: The whole module is marked as integration tests

**Pull request merged**